### PR TITLE
feat(layer): use local host to install requirements

### PR DESCRIPTION
**Pull request by oconpa**
Fixes #

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.

### CODE REVIEW

File: src/patterns/gen-ai/aws-langchain-common-layer/index.ts [hunk 1]
```diff
@@ -38,6 +38,14 @@ export interface LangchainLayerProps{
    * @default - none
    */
   readonly autoUpgrade?: boolean;
+  /**
+   * Optional: Local compute will be used when installing requirements.txt.
+   * By default if this is not true, a docker container will be spun up to install requirements.
+   * The string value will be the python alias used to install requirements.
+   *
+   * @default - none
```
Comment (scottschreckengaust) [inline src/patterns/gen-ai/aws-langchain-common-layer/index.ts:46]:
```suggestion
   * @default - false
```

File: src/patterns/gen-ai/aws-langchain-common-layer/index.ts [hunk 2]
```diff
@@ -38,6 +38,14 @@ export interface LangchainLayerProps{
    * @default - none
```
Comment (scottschreckengaust) [inline src/patterns/gen-ai/aws-langchain-common-layer/index.ts:38]:
```suggestion
   * @default - false
```

File: src/patterns/gen-ai/aws-langchain-common-layer/README.md [hunk 3]
```diff
@@ -183,6 +183,7 @@ Parameters
 | runtime | [lambda.Runtime](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Runtime.html) | ![Required](https://img.shields.io/badge/required-ff0000) | Lambda function runtime compatible with this layer. |
 | architecture | [lambda.Architecture](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Architecture.html)| ![Required](https://img.shields.io/badge/required-ff0000) | Lambda function architecture compatible with this layer. |
 | autoUpgrade | boolean | ![Optional](https://img.shields.io/badge/optional-4169E1) | Add '--upgrade' to pip install requirements.txt. In case of a LangchainCommonLayer, this parameter is not used. |
+| local | "python" or "python3" | ![Optional](https://img.shields.io/badge/optional-4169E1) | Local compute will be used when installing requirements.txt. By default if this is not true, a docker container will be spun up to install requirements. The string value will be the python alias used to install requirements. |
```
Comment (scottschreckengaust) [inline src/patterns/gen-ai/aws-langchain-common-layer/README.md:186]:
```suggestion
| local | "python" or "python3" | ![Optional](https://img.shields.io/badge/optional-4169E1) | Local compute will be used when installing requirements.txt when set. By default, a docker container will be spun up to install requirements. |
```

File: src/patterns/gen-ai/aws-langchain-common-layer/index.ts [hunk 4]
```diff
@@ -38,6 +38,14 @@ export interface LangchainLayerProps{
...
    */
   readonly autoUpgrade?: boolean;
+  /**
+   * Optional: Local compute will be used when installing requirements.txt.
+   * By default if this is not true, a docker container will be spun up to install requirements.
+   * The string value will be the python alias used to install requirements.
+   *
+   * @default - false
```
Comment (scottschreckengaust) [inline src/patterns/gen-ai/aws-langchain-common-layer/index.ts:46]: I was incorrect at the use, and had inadvertently marked this to force to boolean, it was the `autoUpgrade`:
```suggestion
   * @default - none
```

**Review by scottschreckengaust — DISMISSED** [review]
LGTM

File: src/patterns/gen-ai/aws-langchain-common-layer/index.ts [hunk 5]
```diff
@@ -35,9 +35,17 @@ export interface LangchainLayerProps{
    * Optional: Add '--upgrade' to pip install requirements.txt
    * In case of a LangchainCommonLayer, this parameter is not used.
    *
-   * @default - none
+   * @default - false
    */
   readonly autoUpgrade?: boolean;
+  /**
+   * Optional: Local compute will be used when installing requirements.txt.
+   * By default if this is not true, a docker container will be spun up to install requirements.
```
Comment (scottschreckengaust) [inline src/patterns/gen-ai/aws-langchain-common-layer/index.ts:43]:
```suggestion
   * By default, a docker container will be spun up to install requirements. To override this behavior, use the python alias string of `python` or `python3`
```

**Review by krokoko — APPROVED** [review]

**Review by laithalsaadoon — APPROVED** [review]

**Review by scottschreckengaust — APPROVED** [review]
LGTM

**Pull request merged**